### ISSUE TITLE
Incorrect parsing of names that are also prefixes, conjunctions or suffixes

### ISSUE THREAD

**Issue by GoogleCodeExporter** [gdsaxton/python-nameparser#13]
```
According to the 2000 U.S. Census 
[http://www.census.gov/genealogy/www/data/2000surnames/index.html], the 
following prefixes and titles are also surnames of at least 100 people in the 
U.S.:

surname,count
LE,77453
VAN,12531
LA,4892
DOCTOR,3159
BEN,2838
SAN,1810
MISTER,1215
SA,1187
BON,957
DE,955
DER,905
MAJ,415
BIN,415
ABU,383
DA,359
DI,356
VON,344
ENS,249
DEL,248
SIR,230
COL,179
GEN,155
AND,150
MISS,149
DAL,127
AB,106

Name parsing is hard.  :)
```

Original issue reported on code.google.com by `jayqhac...@gmail.com` on 8 Feb 2012 at 7:45

**Comment by GoogleCodeExporter** [comment]
```
I've run into this issue with the name Ben (an extremely common name). For 
example for the following HumanName:
In [9]: h = HumanName('Ben Jones')

In [10]: h.capitalize()

In [11]: h.first
Out[11]: u'Ben Jones'

In [20]: h.last
Out[20]: u''

Changing the prefixes set fixes this, but I'm not certain this is the optimal 
solution:

In [23]: p = nameparser.constants.PREFICES.copy()

In [24]: p.remove('ben')

In [25]: h = HumanName('Ben Jones', prefixes_c=p)

In [26]: h.capitalize()

In [27]: h.first
Out[27]: u'Ben'

In [28]: h.last
Out[28]: u'Jones'

One alternative would be that if last is an empty string (see 1st example 
above), assume there is no prefix. This is a bit kludgy, but will likely handle 
99% of the cases.
```

Original comment by `mdbec...@gmail.com` on 1 May 2012 at 4:25

**Comment by GoogleCodeExporter** [comment]
Original comment by `dere...@gmail.com` on 7 Oct 2012 at 10:44
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
Issue 24 has been merged into this issue.
```

Original comment by `dere...@gmail.com` on 7 Oct 2012 at 10:45

**Comment by GoogleCodeExporter** [comment]
Original comment by `dere...@gmail.com` on 7 Oct 2012 at 10:53
- Changed title: **Incorrect parsing of names that are also prefixes, conjunctions or suffixes**

**Comment by GoogleCodeExporter** [comment]
```
Issue 18 has been merged into this issue.
```

Original comment by `dere...@gmail.com` on 7 Oct 2012 at 10:55

**Comment by GoogleCodeExporter** [comment]
Original comment by `dere...@gmail.com` on 7 Oct 2012 at 11:01
- Added labels: **Priority-High**
- Removed labels: **Priority-Medium**

**Comment by GoogleCodeExporter** [comment]
```
This should be fixed in v0.2.4
```

Original comment by `dere...@gmail.com` on 11 Feb 2013 at 3:47
- Changed state: **Fixed**
